### PR TITLE
Add watch task to run unit test

### LINKED ISSUE #177: Adding watch/test task
What about adding a gulp task to automatically running the tests when files changed:

```javascript
gulp.task('auto-test', function() {
    gulp.watch(['lib/**', 'test/**'], ['test']);
});
```

### PR DESCRIPTION
**Pull request by zckrs**
fix #177 
Add task and apply `yo node:gulp` on generator.

### CODE REVIEW

File: gulpfile.js [hunk 1]
```diff
@@ -21,7 +21,7 @@ gulp.task('nsp', function (cb) {
 });
 
 gulp.task('pre-test', function () {
-  return gulp.src(['**/*.js', '!**/templates/**'])
+  return gulp.src(['generators/**/*.js', '!**/templates/**'])
```
Comment (SBoudrias) [inline gulpfile.js]: We actually want to cover every js files in here. It's useful to know when there's dead code in test code, etc.

Comment (zckrs) [inline gulpfile.js]: Done.

File: package.json [hunk 2]
```diff
@@ -14,7 +14,8 @@
   "author": "Yeoman team",
   "repository": "yeoman/generator-node",
   "scripts": {
-    "test": "mocha test -R spec"
+    "test": "gulp",
```
Comment (hemanth) [inline package.json]: :+1:

**Comment by SBoudrias** [comment]
Thanks!

**Comment by hemanth** [comment]
Ok, time to delete the other branch related to #177

**Pull request merged**